### PR TITLE
Adding changes to upload to Ansible Hub

### PR DESCRIPTION
**Pull request by hcherukuri**

### CODE REVIEW

**Comment by hcherukuri** [comment]
On local molecule tests are good. For now reason there is an issue with redhat_csp_download role on github actions.

**Comment by hcherukuri** [comment]
https://pastebin.com/GpwKm3F8
https://pastebin.com/MDPNFtmA

**Comment by rpelisse** [comment]
@hcherukuri Am I right to assume this PR includes (and thus supersedes) [PR24](https://github.com/ansible-middleware/jws-ansible-playbook/pull/24)? 

The CI run failing puzzles me however:
```FAILED! => {"reason": "couldn't resolve module/action 'community.general.xml'. ```

This does not feel like something that could happen because of the namespace change... Did you get a chance to look into it?

**Comment by hcherukuri** [comment]
> @hcherukuri Am I right to assume this PR includes (and thus supersedes) [PR24](https://github.com/ansible-middleware/jws-ansible-playbook/pull/24)?
> 
> The CI run failing puzzles me however: `FAILED! => {"reason": "couldn't resolve module/action 'community.general.xml'. `
> 
> This does not feel like something that could happen because of the namespace change... Did you get a chance to look into it?

@rpelisse I and @sabre1041  took a look into this and  for some reason community.general is not getting loaded into CI. Molecule tests work fine on the local so we have added "ansible-galaxy collection install -r molecule/default/requirements.yml" in CI to fix this issue. Please take a look into the PR and let me know if you see any issues.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,20 +26,22 @@ jobs:
...
+          pip install yamllint 'molecule[docker]~=3.5.2' ansible-core flake8 ansible-lint voluptuous
 
-      - name: Create default collection path symlink
-        run: |
-          mkdir -p /home/runner/.ansible
-          ln -s /home/runner/work/middleware_automation/jws /home/runner/.ansible/collections
+      # - name: Create default collection path symlink
+      #   run: |
+      #     mkdir -p /home/runner/.ansible
+      #     ln -s /home/runner/work/redhat/jws /home/runner/.ansible/collections
```
Comment (rpelisse) [inline .github/workflows/ci.yml:35]: Should we not remove those commented lines?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -26,20 +26,22 @@ jobs:
...
       - name: Run molecule test
-        run: molecule test
+        run: |
+          ansible-galaxy collection install -r molecule/default/requirements.yml
+          molecule test
         env:
           PY_COLORS: "1"
           ANSIBLE_FORCE_COLOR: "1"
-        working-directory: ./ansible_collections/middleware_automation/jws
+        working-directory: ./ansible_collections/redhat/jws
```
Comment (rpelisse) [inline .github/workflows/ci.yml:47]: This is the third occurrences of 'ansible_collections/redhat/jws' in this file... Can we maybe use some sort of variable?

File: ansible.cfg [hunk 3]
```diff
@@ -1,7 +1,7 @@
 [defaults]
 host_key_checking = False
 retry_files_enabled = False
-callback_whitelist = profile_tasks
+#callback_whitelist = profile_tasks
```
Comment (rpelisse) [inline ansible.cfg:4]: Maybe remove it rather than comment it?

File: playbook.yml [hunk 4]
```diff
@@ -37,6 +37,7 @@
     - redhat_csp_download
     - jws
     - common_criteria
+
```
Comment (rpelisse) [inline playbook.yml:40]: I would remove this, just for the sake of having a clean commit.

File: roles/common_criteria/meta/main.yml [hunk 5]
```diff
@@ -1,3 +1,4 @@
 ---
+# Collections list
 collections:
-  - community.general
+  - community.general
```
Comment (rpelisse) [inline roles/common_criteria/meta/main.yml:4]: I would also remove this, just for the sake of having a clean commit.

**Review by rpelisse — APPROVED** [review]
@hcherukuri I did a few comment, but overall, great work! I'm happy this is finally passing CI!
@sabre1041 any thoughts on this PR?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (hcherukuri) [reply .github/workflows/ci.yml:42]: I would rather keep it for clear understanding.

**Review by rpelisse — APPROVED** [review]

**Pull request merged**